cmd/tailscale/cli/whois.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/peterbourgon/ff/v3/ffcli"
+)
+
+var whoisCmd = &ffcli.Command{
+	Name:       "whois",
+	ShortUsage: "whois [--json] [ip|ip:port]",
+	ShortHelp:  "Show the machine and user associated with a Tailscale IP",
+	LongHelp: strings.TrimSpace(`
+	'tailscale whois' shows the machine and user associated with a Tailscale IP.
+	`),
+	UsageFunc: usageFunc,
+	Exec:      runWhoIs,
+	FlagSet: func() *flag.FlagSet {
+		fs := newFlagSet("whois")
+		fs.BoolVar(&whoIsArgs.json, "json", false, "output in JSON format")
+		return fs
+	}(),
+}
+
+var whoIsArgs struct {
+	json bool
+}
+
+func runWhoIs(ctx context.Context, args []string) error {
+	if len(args) > 1 {
+		fmt.Println(args)
+		return errors.New("too many arguments, expected at most one peer")
+	} else if len(args) == 0 {
+		return errors.New("missing argument, expected one peer")
+	}
+	who, err := localClient.WhoIs(ctx, args[0])
+	if err != nil {
+		return err
+	}
+	if whoIsArgs.json {
+		ec := json.NewEncoder(Stdout)
+		ec.SetIndent("", "  ")
+		ec.Encode(who)
+		return nil
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 10, 5, 5, ' ', 0)
+	fmt.Fprintf(w, "Machine:\n")
+	fmt.Fprintf(w, "  Name:\t%s\n", strings.TrimSuffix(who.Node.Name, "."))
+	fmt.Fprintf(w, "  ID:\t%s\n", who.Node.StableID)
+	if who.Node.IsTagged() {
```

Comment:
Nit: move down to before 65? Keep `!tagged` and `tagged` blocks together

Reply:
I tried doing that but then the flow looks awkward. As written `Tags` show up under `Machine`, by moving it later they would show up at the top level

Reply:
What if you just move it to be under `Addresses`, but only add the user section if not tagged?

```
if who.Node.IsTagged() {
	fmt.Fprintf(w, "  Tags:\t%s\n", strings.Join(who.Node.Tags, ", "))
} else {
        fmt.Fprintln(w, "User:")
        ...
}
```

Reply:
sure

Reply:
done